app/controllers/scan_analytics_controlller.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ScanAnalyticsController < ApplicationController
+  helper ProjectsHelper
+
+  before_action :set_project_or_fail
+  before_action :analytics_data, only: :index
+
+  def index
+    return if @analytics.blank?
+
+    params[:code_set_id] = params[:code_set_id].nil? ? @analytics.first.code_set_id : params[:code_set_id]
```

Comment:
This can be rewritten to:
```
  params[:code_set_id] = params[:code_set_id] ? params[:code_set_id] :  @analytics.first.code_set_id
```
We do not need to check for `nil`.